internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -2804,6 +2806,28 @@ func (r *Reconciler) reconcilePGBackRestCronJob(
 		naming.PGBackRestCronJobLabels(cluster.Name, repo.Name, backupType),
 	)
 	objectmeta := naming.PGBackRestCronJob(cluster, backupType, repo.Name)
+
+	// Look for an existing CronJob by the associated Labels. If one exists,
+	// update the ObjectMeta accordingly.
+	for _, cronjob := range cronjobs {
+		// ignore CronJobs that are terminating
+		if cronjob.GetDeletionTimestamp() != nil {
+			continue
+		}
+		clusterName := cronjob.GetLabels()[naming.LabelCluster]
+		cronjobBackupType := cronjob.GetLabels()[naming.LabelPGBackRestCronJob]
+		repoName := cronjob.GetLabels()[naming.LabelPGBackRestRepo]
+
+		if clusterName == cluster.Name &&
+			cronjobBackupType == backupType &&
+			repoName == repo.Name {
```

Comment:
⛏️ Do we get anything with the extra variables? The following seems more direct to me. 🤷🏻 

```suggestion
		if cronjob.Labels[naming.LabelCluster] == cluster.Name &&
			cronjob.Labels[naming.LabelPGBackRestCronJob] == backupType &&
			cronjob.Labels[naming.LabelPGBackRestRepo] == repo.Name {
```

Reply:
Not really. When I first started this, the variable names were based off of the label names, not the values, so it made a bit more sense. Now that it's refactored, I agree it makes sense to simplify this. I'll push this change before merging.